src/lib/queue_worker.js

```diff
@@ -113,7 +119,7 @@ QueueWorker.prototype._resetTask = function(taskRef, deferred) {
       } else {
         var errorMsg = 'reset task errored too many times, no longer retrying';
         logger.debug(self._getLogEntry(errorMsg), error);
-        deferred.reject(errorMsg);
+        deferred.reject(new Error(errorMsg));
```

Comment:
I am a fan of using `Error` because it allows you to attach other
properties, etc. You could put the number of errors, etc. Also could create
custom errors for these scenarios. But yes, a string works too.

On Tue, Jun 16, 2015 at 10:20 AM, Matthew Tse <notifications@github.com>
wrote:

> In src/lib/queue_worker.js
> <https://github.com/firebase/firebase-queue/pull/24#discussion_r32546129>:
>
> > @@ -113,7 +119,7 @@ QueueWorker.prototype._resetTask = function(taskRef, deferred) {
> >        } else {
> >          var errorMsg = 'reset task errored too many times, no longer retrying';
> >          logger.debug(self._getLogEntry(errorMsg), error);
> > -        deferred.reject(errorMsg);
> > +        deferred.reject(new Error(errorMsg));
>
> What's the reason for this? Passing a plain string into reject() should
> work as well correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/firebase/firebase-queue/pull/24/files#r32546129>.
>